File: aat-android/src/main/java/ch/bailu/aat/description/TotalStepsDescription.java
package ch.bailu.aat.description;

import android.content.Context;

import ch.bailu.aat_lib.description.ContentDescription;
import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.gpx.attributes.GpxAttributes;
import ch.bailu.aat_lib.gpx.attributes.StepCounterAttributes;
import ch.bailu.aat_lib.resources.Res;

public class TotalStepsDescription extends ContentDescription {

    private String value = VALUE_DISABLED;
    private final String unit;
    private final String label;


    public TotalStepsDescription(Context c) {
        unit = Res.str().sensor_step_total_unit();
        label = Res.str().sensor_step_total();
    }

    @Override
    public String getValue() {
        return value;
    }

    @Override
    public String getLabel() {
        return label;
    }

    @Override
    public String getUnit() {
        return unit;
    }


    @Override
    public void onContentUpdated(int iid, GpxInformation info) {

        GpxAttributes attr = info.getAttributes();

        if (attr.hasKey(StepCounterAttributes.KEY_INDEX_STEPS_TOTAL)) {
            value = info.getAttributes().get(StepCounterAttributes.KEY_INDEX_STEPS_TOTAL);
        } else {
            value = VALUE_DISABLED;


        }
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/TotalStepsDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.gpx.attributes.StepCounterAttributes
import ch.bailu.aat_lib.resources.Res

class TotalStepsDescription : ContentDescription() {
    private var value = VALUE_DISABLED
    private val unit: String = Res.str().sensor_step_total_unit()
    private val label: String = Res.str().sensor_step_total()

    override fun getValue(): String {
        return value
    }

    override fun getLabel(): String {
        return label
    }

    override fun getUnit(): String {
        return unit
    }

    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        val attr = info.getAttributes()
        value = if (attr.hasKey(StepCounterAttributes.KEY_INDEX_STEPS_TOTAL)) {
            info.getAttributes()[StepCounterAttributes.KEY_INDEX_STEPS_TOTAL]
        } else {
            VALUE_DISABLED
        }
    }
}
